matron/src/device/device_common.h

```diff
@@ -9,14 +9,18 @@ typedef enum {
     DEV_TYPE_HID = 1,
     // raw midi devices
     DEV_TYPE_MIDI = 2,
-    DEV_TYPE_MIDI_VIRTUAL = 3,
     // usbmodem (crow)
-    DEV_TYPE_CROW = 4,
+    DEV_TYPE_CROW = 3,
+    // place all virtual devices (devices without device files to monitor) below
+    DEV_TYPE_MIDI_VIRTUAL = 4,
     // counter - unused, don't remove
     DEV_TYPE_COUNT,
     DEV_TYPE_INVALID
 } device_t;
 
+// maximum device_t value for devices which have corresponding device files
+#define DEV_TYPE_COUNT_PHYSICAL 4
```

Comment:
i think DEV_TYPE_COUNT_PHYSICAL could be also declared in the device type enum.

Reply:
I wasn't sure of that; I thought if I did it would throw off the implicit count which `DEV_TYPE_COUNT` is set to.

Reply:
...to be clear i'm not loving this fix, it is more fragile than i'd like but it seemed the least bad of the options i tried.

the thing which broke when i changed the enum around was related to this static array:

Reply:
FWIW - I worked around that static array on Line 44 in my hacks to device_monitor for teensy (ACM) devices - [mentioned in this issue](https://github.com/monome/norns/issues/955#issuecomment-561963162)

I will test tomorrow if this works with the virtual midi devices or not. 

Don't know if this helps or not, but thought I'd mention it again.